### PR TITLE
Issue #4641: React ErrorBoundary styles should be loaded automatically

### PR DESCRIPTION
**Pull request by deltaepsilon**
## Summary

Addresses issue #4641 

The `<ErrorBoundary>` component exported from `@highlight-run/react` did not have self-contained styles. The user had to manually import `@highlight-run/react/dist/index.css`.

## How did you test this change?

We've added an `<ErrorBoundaryButton />` to the `nextjs` test project, and wired it up to the new `<ErrorBoundary>` component. Run `yarn e2e:nextjs` from the project root to spin up the server on `localhost:3005`. Click the "throw an error" button to trigger.

image

image

## Future considerations

Issue #4288 is a redesign of the `<ErrorBoundary showDialog>` dialog. I'll reconsider the font situation during that effort.

## Are there any deployment considerations?

n/a

### CODE REVIEW

File: sdk/highlight-react/build.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const esbuild = require('esbuild')
+const cssModulesPlugin = require('esbuild-css-modules-plugin')
+const isWatch = process.argv.includes('--watch')
+const pkg = require('./package.json')
+const config = {
+	logLevel: 'info',
+	entryPoints: ['src/index.tsx'],
+	bundle: true,
+	plugins: [cssModulesPlugin()],
```
Comment (deltaepsilon) [inline sdk/highlight-react/build.js:9]: I needed the `cssModulesPlugin` and plugins can only be used with esbuild's `node` api... so bye-bye `tsup`, hello `build.js`

File: sdk/highlight-react/src/components/ReportDialog/styles.d.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+declare module '*.module.css' {
+	const classes: { [key: string]: string }
+	export default classes
+}
```
Comment (ccschmitz) [inline sdk/highlight-react/src/components/ReportDialog/styles.d.ts:4]: Nit, but perhaps we could use https://www.npmjs.com/package/typed-scss-modules to generate these types in the future if we stick with using CSS modules. I think that's what we use in the main app to keep types in sync for our styles.

File: .yarnrc.yml [hunk 3]
```diff
@@ -3,5 +3,7 @@ nodeLinker: node-modules
 plugins:
     - path: .yarn/plugins/@yarnpkg/plugin-workspace-tools.cjs
       spec: '@yarnpkg/plugin-workspace-tools'
+    - path: .yarn/plugins/@yarnpkg/plugin-interactive-tools.cjs
+      spec: '@yarnpkg/plugin-interactive-tools'
```
Comment (ccschmitz) [inline .yarnrc.yml:7]: Would be cool to show this off to the team at some point! Maybe worth mentioning in #demo.

**Review by ccschmitz — APPROVED** [review]
Code + click test look great! Nice work on this, @deltaepsilon 👏

File: sdk/highlight-react/package.json [hunk 4]
```diff
@@ -19,6 +19,9 @@
 		"@types/node": "^18.7.16",
 		"@types/react": "^18.0.28",
 		"@types/react-dom": "^18.0.11",
+		"classnames": "^2.3.2",
+		"css-tree": "^2.3.1",
```
[the remark this answers is not in the document]
Comment (deltaepsilon) [reply sdk/highlight-react/package.json:23]: I'll drop this. It was recommended by another package and I forgot to remove it.

File: .yarnrc.yml (hunk 3 above)
Comment (deltaepsilon) [reply .yarnrc.yml:7]: Done!

**Pull request merged**